### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by SAZZAD1Q2**
Set unit tests for the files operate.js and calculate.js using Jest.
Create unit tests for React components:
Use Jest snapshots to test the components.
Use React Testing Library to simulate user interaction.

I am working alone in this week. Microverse permitted me to do so. Here is the screenshot of their mail: 
Screenshot_2

Screenshot_3

Here is the screenshot of my dashboard :   Screenshot_4

### CODE REVIEW

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @SAZZAD1Q2 👋,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Status: Invalid Code Review Request

<div><br /></div>

You have submitted a project, that requires pair-programming. Kindly ensure your partner contributes.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by AlphaNtihinduka — COMMENTED** [review]
Hi @SAZZAD1Q2 ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
# Invalid Code Review Request ❌ 
was-that-wrong-icegif

- [ ] You have submitted a project that is a pair-programming. I have seen that you are working alone this week. However, I suggest that you take the screenshot of the email from Microverse and add it to the PR description.
- [ ] In addition, it has been communicated that the Errors with `webhint` have been fixed. Kindly ensure to add the webhint error checkers ☺️.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready pray

Cheers and Happy coding!clapclapclap

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Review by AlphaNtihinduka — COMMENTED** [review]
Hi @SAZZAD1Q2 ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
# Invalid Code Review Request ❌ 
was-that-wrong-icegif

- [ ] You have submitted a project that is a pair-programming. I have seen that you are working alone this week. However, I suggest that you take the screenshot of the email from Microverse and add it to the PR description.
- [ ] In addition, it has been communicated that the Errors with `webhint` have been fixed. Kindly ensure to add the webhint error checkers ☺️.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready pray

Cheers and Happy coding!clapclapclap

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/_test_/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import renderer from 'react-test-renderer';
+import Home from '../components/Home';
+
+describe('Home', () => {
+  test('should render the component', () => {
+    const component = renderer.create(<Home />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (iambenkis) [inline src/_test_/Home.test.js:11]: - [ ] Well done making most of the tests passed 👏  , however there are still some you still need to fix . I know they may be working on your computer but always remember that most of the product you will make will be running on clients computers 🆗  , Kindly fix the unpassed tests to meet the clients need 😄 😄 😄 
- [ ] Also I noticed that there is no snapshot test running , kindly fix that also 👍 

Capture

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi Team @SAZZAD1Q2 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  Unit tests are passing 

## Suggested changes

- [ ] Kindly provide information about tests , how to install them , how to run them as they are important part of you project. The usage sub-section should be present in that case 👍 

Capture
 

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/_test_/Home.test.js (hunk 1 above)
Comment (SAZZAD1Q2) [reply src/_test_/Home.test.js:11]: Dear @iambenkis,
Thank you very much for reviewing my pr. I appreciate your informative advice. The test for the project is working well on my computer. Will you suggest to me how to test for all computers? Besides I used the snapShots test for my project.
Thank you once again.
Md Sazzad Hossain

**Review by DeliceLydia — APPROVED** [review]
## Hi @SAZZAD1Q2, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**